xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -232,7 +232,11 @@ void CDirectoryProvider::Announce(ANNOUNCEMENT::AnnouncementFlag flag, const cha
          (std::find(m_itemTypes.begin(), m_itemTypes.end(), InfoTagType::AUDIO) == m_itemTypes.end())))
       return;
 
-    if (flag & ANNOUNCEMENT::Player)
+    if ((flag & ANNOUNCEMENT::Player) &&
+        ((CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() == WINDOW_FULLSCREEN_VIDEO) ||
```

Comment:
The check for these four windows (and often missing a window) occurs in many places now, e.g. [here](https://github.com/xbmc/xbmc/blob/826c8cf94869dd995db7304088f491d4367ca36d/xbmc/PlayListPlayer.cpp#L957), [here](https://github.com/xbmc/xbmc/blob/826c8cf94869dd995db7304088f491d4367ca36d/xbmc/Application.cpp#L2104), [here](https://github.com/xbmc/xbmc/blob/826c8cf94869dd995db7304088f491d4367ca36d/xbmc/Application.cpp#L3334), [here](https://github.com/xbmc/xbmc/blob/826c8cf94869dd995db7304088f491d4367ca36d/xbmc/Application.cpp#L4124), aaaaaaand got tired of searching :) but you get the idea. cleaning up this duplication by factoring it out would be a welcome change.